src/test/ui/resolve/issue-101749.stderr

```diff
@@ -0,0 +1,14 @@
+error[E0433]: failed to resolve: use of undeclared crate or module `rect1`
+  --> $DIR/issue-101749.rs:19:20
+   |
+LL |     println!("{}", rect1::area());
+   |                    ^^^^^ use of undeclared crate or module `rect1`
+   |
+help: `rect1` is not a crate or module, maybe you meant to call instance method
```

Comment:
I think this suggestion is emitted if rect1 does not have `area` method. Is it possible to check if rect has it before creating suggestion?

Given the following code:

```rust
// run-rustfix
struct Rectangle {
    width: i32,
    height: i32,
}
impl Rectangle {
    fn new(width: i32, height: i32) -> Self {
        Self { width, height }
    }
}

fn main() {
    let width = 3;
    let height = 4;
    let rect1 = Rectangle::new(width, height);
    println!("{}", rect1::area());
    //~^ ERROR failed to resolve: use of undeclared crate or module `rect1`
}
```

Current output is:

```
error[E0433]: failed to resolve: use of undeclared crate or module `rect1`
  --> src/main.rs:27:20
   |
27 |     println!("{}", rect1::area());
   |                    ^^^^^ use of undeclared crate or module `rect1`
   |
help: `rect1` is not a crate or module, maybe you meant to call instance method
   |
27 |     println!("{}", rect1.area());
   |                    ~~~~~~

For more information about this error, try `rustc --explain E0433`.
```

Reply:
You are right.
This need to delay the suggestion to be emit in type checking phase, I will have a try.

Reply:
Spent some time to trying this, I think it's complicated to make it work. We need to find the local binding of and `ty` of the ident, then to check whether it has the method.

See my branch:
https://github.com/chenyukang/rust/commit/9cf2a602c774666c7262e75059370ebb65b4b4c1

Not sure whether worth it for a corner case of suggestion.

Reply:
Adding a field like the following for storing impl items might help you.

https://github.com/rust-lang/rust/blob/4891d57f7aab37b5d6a84f2901c0bb8903111d53/compiler/rustc_resolve/src/lib.rs#L1026

Reply:
I continue with my previous solution, deply to checking method in type checking phase, now it works by checking whether the method name is valid (may need more tweak about the arguments checking).

I have a question for my current code, how can we find the binding according to a `ident`(then find out the `type` of it), 

My code for this part is:
https://github.com/rust-lang/rust/pull/101975/files#diff-8dea1c54efc225f1b3a27454cbf2fc50f016073f51319b27d1b834bee7c5419eR1435

one obvious issue is it can only handle the `let` expression, I want our suggestion works also for other scenarios, such as the binding is introduced by function arguments.

Reply:
@compiler-errors do you know any better solution for this?